src/gpodder/gtkui/main.py

```diff
@@ -3070,6 +3090,16 @@ def on_item_toggle_lock_activate(self, unused, toggle=True, new_value=False):
         self.on_selected_episodes_status_changed()
         self.play_or_download()
 
+    def on_episode_lock_action(self, action, *params):
+        new_state = not action.get_state().get_boolean()
+        for episode in self.get_selected_episodes():
+            episode.mark(is_locked=new_state)
+        self.on_selected_episodes_status_changed()
+        self.play_or_download()
+        action.change_state(GLib.Variant.new_boolean(new_state))
```

Comment:
Could the state be set when opening the menu instead of when closing it? Same applies for the "new" toggle below.

Reply:
The state is set (look for 'New and Archive state' in the code) before the popover is shown.

Is there any harm in setting the new state here? The episodes remain selected after the popover closes, so the state is still valid.

Reply:
Ok, I see that state is set when opened, but what is the purpose of setting it immediately before the popover is closed? I've tested and the checkboxes don't update in the split second before it closes.

Reply:
The episodes remain selected, so the state of the action now represents their current state. There could be some other UI element which shows the state of the action, in a toolbar for example (currently there's not), which would be then updated.

Reply:
Okay.